### PR TITLE
Topic/tempo clock is running

### PR DESCRIPTION
**Pull request by jamshark70**
Purpose and Motivation
----------------------

It's kind of a no-brainer to be able to ask a TempoClock if it's still running -- it might have been stopped, but in the main class library as it is now, you have no way to know, except to try to schedule something on it and catch an error (!).

So I propose a very simple `isRunning` method. Actually, I've had this in my extensions for years. It works.

Plus documentation.

Plus a unit test... but then I thought, there are some other things about TempoClock (correct rate of advancing through time, handling of meter changes) that are probably not tested anywhere else. So I added a bunch of other useful tests.

Types of changes
----------------

- Documentation (non-code change which corrects or adds documentation for existing features)
- New feature (non-breaking change which adds functionality)

Checklist
---------

- [x] All previous tests are passing (well, I did try to run the entire test suite, but TestUGen_Duty locked up the system and forced me to hard reboot... so to be completely honest, I don't know, but I don't think that's the fault of this PR)
- [x] Tests were updated or created to address changes in this PR, and tests are passing
- [x] Updated documentation, if necessary
- [x] This PR is ready for review

### CODE REVIEW

File: testsuite/classlibrary/TestTempoClock.sc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		}.fork(clock);
+		cond.hang;
+		// note: floating point math, == is not safe
+		this.assert((then - now absdif: 0.001) < 1e-10, "TempoClock 1000 beats/sec, 1 beat should be 0.001 seconds");
+	}
+
+	test_beats2secs_validConversion {
+		this.assert(
+			clock.beats2secs(clock.beats + 1) - SystemClock.seconds absdif: 0.001 < 1e-10,
+			"TempoClock 1000 beats/sec, 'beats2secs' for now + 1 beat should be 0.001 seconds later"
```
Comment (brianlheim) [inline testsuite/classlibrary/TestTempoClock.sc]: use assertFloatEquals

File: testsuite/classlibrary/TestTempoClock.sc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		)
+	}
+
+	test_secs2beats_validConversion {
+		this.assert(
+			clock.secs2beats(SystemClock.seconds + 1) - clock.beats absdif: 1000.0 < 1e-10,
+			"TempoClock 1000 beats/sec, 'secs2beats' for now + 1 second should be 1000 beats later"
```
Comment (brianlheim) [inline testsuite/classlibrary/TestTempoClock.sc]: use assertFloatEquals

File: testsuite/classlibrary/TestTempoClock.sc [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		)
+	}
+
+	test_defaultMeter_unchanged {
+		this.assert(
+			clock.beatsPerBar == 4,
+			"A new TempoClock should have 4 beats per bar by default"
```
Comment (brianlheim) [inline testsuite/classlibrary/TestTempoClock.sc]: use assertEquals, change method name to `test_beatsPerBar_defaultIs4` and remove message

Comment (brianlheim) [reply testsuite/classlibrary/TestTempoClock.sc]: may want to turn this into a test for multiple properties of a default constructed tempoclock if there are other things you could check

File: testsuite/classlibrary/TestTempoClock.sc [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	// called after each test
+	tearDown {
+		clock.stop;
+	}
+
+	test_isRunning {
+		var beforeStop = clock.isRunning, afterStop;
+		clock.stop;
+		afterStop = clock.isRunning;
+		this.assert(beforeStop and: { afterStop.not }, "TempoClock:isRunning should be true before 'stop', and false afterward");
```
[the remark this answers is not in the document]
Comment (jamshark70) [reply testsuite/classlibrary/TestTempoClock.sc]: No disagreement with any of these suggestions. Done.

**Comment by brianlheim** [comment]
Please wait until after 3.10.1 to merge

**Pull request merged**